Add Discord integration docs (#7645)
getsentry/sentry-docs: src/docs/product/integrations/notification-incidents/discord/index.mdx

---
title: Discord
sidebar_order: 1
description: "Learn more about Sentry's Discord integration and how you can get notified and manage issues right in Discord."
---

> 

Reviewer: ```suggestion

<Include name="feature-available-for-user-group-early-adopter.mdx" />

```

Get notified and manage issues right in Discord with Sentry's Discord integration.

## Install

<Note>

Sentry owner, manager, or admin permissions are required to install this integration.

</Note>

1. In [sentry.io](https://sentry.io), nagivate to **Settings > Integrations > Discord**.

1. Click "Add Installation".

   ![Install Discord integration](discord-integration-details.png)

> 1. A pop up window will open. If you're not logged in to Discord in your browser, you'll need to log in. If you're already logged in, you won't see this.

Reviewer 2: Can we make this a bit less verbose? just a suggestion - 

```suggestion
1. Log in to Discord, if necessary. If you're already logged in, the login popup will not be displayed.```

   ![Log in to Discord](discord-login.png)

1. Use the dropdown menu to select the Discord server you'd like to add Sentry to and press continue.

   <Note>
     You will need to have the "Manage Server" permission enabled for your
     Discord account in the server you're trying to add the bot to.
   </Note>

   ![Select server](discord-server-select.png)

1. Finally, once the server is selected, click authorize. The integration will be created and the Sentry Discord bot will join the selected server.

Your Discord integration is now available to all projects in your Sentry organization.

In the next section, we'll walk through configuring Discord notifications.

## Configure

> Currently our Discord integration supports just issue alert rules, but metric alerts and personal notifications are planned for future updates to the integration.

Reviewer 2: ```suggestion
Currently our Discord integration supports issue alert rules, with metric alerts and personal notifications planned for future updates to the integration.
```

### Issue Alerts

> Set up an issue alert by going to **Alerts** and clicking "Create Alert". "Issues" is the default selection so you can just click "Set Conditions".

Reviewer 2: ```suggestion
Set up an issue alert by going to **Alerts** and clicking "Create Alert".  Select "Issues" and then click "Set Conditions".
```

![Create issue alert](discord-add-issue-alert.png)

> From here you can configure your issue alert conditions as desired. Discord is then added by selecting "Send a Discord notification" in the actions dropdown.

Reviewer 2: ```suggestion
Configure your issue alert conditions as desired and then add Discord by selecting "Send a Discord notification" in the actions dropdown.
```

Select the Discord server and channel you'd like to send the alert to. You can optionally specify tags here to include in the notification.

![Discord issue alert action settings](discord-action-form.png)

> The channel field must be filled in with a Discord channel ID, _not_ a channel name (this is because multiple Discord channels can share a name). To get this channel ID, you must enable Discord's Developer Mode (Settings -> Advanced -> Developer Mode).

Reviewer 2: We can exclude some of the finer implementation details. Also, include a link to Discord docs on how to enable developer mode.

```suggestion
The channel field must be filled in with a Discord channel ID, _not_ a channel name. To get this channel ID, enable Discord's Developer Mode (Settings -> Advanced -> Developer Mode).
```

![Discord Developer Mode Setting](discord-developer-mode.png)

When Developer Mode is enabled, you'll be able to right click a channel and copy the channel's ID.

> Once the channel ID is added, you can test your configuration by clicking "Send Test Notification".
> 
> If everything is working properly, you'll get a notification in Discord.

Reviewer 2: ```suggestion
Once the channel ID is added, you can test your configuration by clicking "Send Test Notification" and seeing the notification appear in Discord.
```

![Discord issue alert](discord-issue-alert.png)

### Identity Linking

Having your Discord account linked to your Sentry account means that you will be able to resolve, archive, and assign issues right in Discord.

If you were the user who installed the Discord bot to your server, your identity should have been linked during the install process. If you were not the installer or want to change the Discord account linked to your Sentry account, you will need to use our Discord bot's `/link` and `/unlink` commands to link or unlink your accounts.

To link your Discord account to your Sentry account, type `/link` in a Discord channel that the Sentry bot has access to. The bot will reply with a link that takes you to our identity linking page.

> ![Discord identity linking pag](discord-identity-link-page.png)

Reviewer 2: ```suggestion
![Discord identity linking page](discord-identity-link-page.png)
```

Author: good catch

> To access this page, you must be signed in to the desired Sentry account in the browser you are trying to open the link in.
> 
> If you see the page as pictured above though, simply click the link button, and your accounts will be linked.
> 
> The process for unlinking is identical.

Reviewer 2: ```suggestion
To access this page, you must be signed in to the desired Sentry account in the browser you are trying to open the link in. 

Click the link button pictured above to link your accounts. Unlinking your identity follows a similar process, with the `/unlink` command.
```

> If you were the user who installed the Discord bot to your server, your identity should have been linked during the install process. If you were not the installer or want to change the Discord account linked to your Sentry account, you will need to use our Discord bot's `/link` and `/unlink` commands to link or unlink your accounts.
> 
> To link your Discord account to your Sentry account, type `/link` in a Discord channel that the Sentry bot has access to. The bot will reply with a link that takes you to our identity linking page.
> 
> ![Discord identity linking pag](discord-identity-link-page.png)
> 
> To access this page, you must be signed in to the desired Sentry account in the browser you are trying to open the link in.
> 
> If you see the page as pictured above though, simply click the link button, and your accounts will be linked.
> 
> The process for unlinking is identical.

Reviewer: ```suggestion
If you’re the user who installed the Discord bot to your server, your identity was linked during the install process. You can use the `/link` command to verify this. The bot will tell you which Sentry account your Discord account is linked to.

All users in any Discord server with the Sentry bot can use the `/link` command to link their accounts. If the account isn’t linked yet, the bot will reply with a link to our identity linking page. Make sure you’re signed into the desired Sentry account in your browser when you open the link. Click "Link with Discord" to link your accounts. 

![Discord identity linking pag](discord-identity-link-page.png)

To unlink your accounts, use the `/unlink` command. Follow the link and click "Unlink with Discord”" 
```
